include/alpaka/kernel/TaskKernelGpuHipRt.hpp

```diff
@@ -76,14 +76,16 @@ namespace alpaka
 #if BOOST_ARCH_PTX && (BOOST_ARCH_PTX < BOOST_VERSION_NUMBER(2, 0, 0))
     #error "Cuda device capability >= 2.0 is required!"
 #endif
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wignored-attributes"
+
+#ifndef BOOST_COMP_HIP
+#   pragma clang diagnostic push
+#   pragma clang diagnostic ignored "-Wignored-attributes"
                     static_assert(
                         std::is_same<typename std::result_of<
                             TKernelFnObj(acc::AccGpuHipRt<TDim, TIdx> const &, TArgs const & ...)>::type, void>::value,
                         "The TKernelFnObj is required to return void!");
-#pragma clang diagnostic pop
-
+#   pragma clang diagnostic pop
+#endif
```

Comment:
Is the `pragma` unsupported or the `static_assert`? I would like to keep the `static_assert` or at least add a comment why this does not work.

Reply:
The evaluation of the static assert expression failed but I now get it solved.

```
                    static_assert(
                        std::is_same<
                            decltype(kernelFnObj(
                                alpaka::core::declval<acc::AccGpuHipRt<TDim, TIdx> const>(),
                                args...)),
                        void>::value,
                        "The TKernelFnObj is required to return void!");
```